File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
    int firstOccurance(vector<int>& nums, int target, int n){
        int start = 0;
        int end = n-1;
        int firstPostion = -1;
        
        while(start <= end){
          int mid = start + (end-start)/2;
            if(nums[mid] == target){
                firstPostion = mid;
                end = mid - 1;
            }
            else if(nums[mid] > target){
                end = mid - 1;
            }
            else{
                start = mid + 1;
            }
        }
        return firstPostion;
    }
    int lastOccurance(vector<int>& nums, int target, int n){
        int start = 0;
        int end = n-1;
        int LastPostion = -1;
        
        while(start<=end){
          int mid = start + (end-start)/2;
            if(nums[mid] == target){
                LastPostion = mid;
                start = mid + 1;
            }
            else if(nums[mid] > target){
                end = mid - 1;
            }
            else{
                start = mid + 1;
            }
        }
        return LastPostion;
    }
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n = nums.size();
        return {firstOccurance(nums,target,n),lastOccurance(nums,target,n)};
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] arr = {-1,-1};

        arr[0] = fristOccurance(nums,target);
        arr[1] = lastOccurance(nums,target);
        return arr; 
    }

    public int fristOccurance(int[] nums,int target){
        int start = 0;
        int end = nums.length -1;
        int index = -1;

        while(start <= end){
            int mid = start + end-start/2;
            if(nums[mid] == target){
                index = mid;
                end = mid -1; 
            }
            else if(nums[mid] > target){
                end = mid -1; 
            }
            else{
                start = mid + 1;
            }
        }
        return index;
    }

    public int lastOccurance(int[] nums, int target){
        int start = 0;
        int end = nums.length -1;
        int index = -1;

        while(start <= end){
            int mid = start + (end - start)/ 2;
            if(nums[mid] == target){
                index = mid;
                start = mid+1;
            }
            else if(nums[mid] > target){
                end = mid-1;
            }
            else{
                start = mid + 1;
            }
        }
        return index;
    }
}